mistralrs-core/src/pipeline/hf.rs

```diff
@@ -64,17 +78,58 @@ pub(crate) fn api_get_file(
         info!("Loading `{file}` locally at `{}`", path.display());
         Ok(path)
     } else {
-        api.get(file).map_err(|e| match e {
-            HFHubApiError::RequestError(err) if matches!(*err, ureq::Error::Status(403, _)) => {
-                HFError::AuthorizationError {
-                    model_id: model_id.display().to_string(),
+        // **Check for SIGTERM before starting the request**
+        if SHOULD_TERMINATE.load(AtomicOrdering::SeqCst) {
+            eprintln!("Download aborted due to SIGTERM.");
+            process::exit(1);
+        }
+
+        setup_signal_handler();
+        // **Start download but abort if SIGTERM is received**
+        let mut download_result = Some(thread::spawn({
+            let api = Arc::clone(api);
+            let file = file.to_string();
+            move || api.get(&file)
+        }));
+
+        while !SHOULD_TERMINATE.load(AtomicOrdering::SeqCst) {
+            if download_result.as_ref().is_some_and(|r| r.is_finished()) {
+                if let Some(Ok(result)) = download_result.take().map(|r| r.join()) {
+                    return result.map_err(|e| match e {
+                        HFHubApiError::RequestError(err)
+                            if matches!(*err, ureq::Error::Status(403, _)) =>
+                        {
+                            HFError::AuthorizationError {
+                                model_id: model_id.display().to_string(),
+                            }
+                        }
+                        ee => HFError::HFHubApiError(ee),
+                    });
                 }
             }
-            ee => HFError::HFHubApiError(ee),
-        })
+            thread::sleep(Duration::from_millis(100)); // Polling loop to check SIGTERM
+        }
+
+        eprintln!("Download aborted due to SIGTERM.");
+        process::exit(1);
```

Comment:
See above - libraries in general shouldn't cause processes to exit - that would prevent normal cleanup from the main process.